maitake/src/wait/cell.rs

```diff
@@ -103,23 +141,25 @@ impl WaitCell {
                     "state changed unexpectedly while parking!"
                 );
 
-                if let Some(prev_waker) = prev_waker {
-                    prev_waker.wake();
-                }
-
                 if let Some(waker) = waker {
                     waker.wake();
                 }
 
-                if test_dbg!(state.is(State::CLOSED)) {
-                    wait::closed()
-                } else {
-                    wait::notified(())
-                }
+                // We just went to the closed state, so sorry new waker, shops
+                // closed
+                closed()
             }
         }
     }
 
+    /// Wait to be woken up by this cell.
```

Comment:
an example could be nice, but we can add that later.

it might also be nice to explicitly state that this does _not_ register the waker until the first time the future is polled...